src/pineko/evolve.py

```diff
@@ -158,18 +175,46 @@ def evolve_grid(
     _x_grid, _pids, mur2_grid, _muf2_grid = grid.axes()
     check.check_grid_and_eko_compatible(grid, operators, xif)
     # rotate to evolution (if doable and necessary)
-    if np.allclose(operators.rotations.pids, br.flavor_basis_pids):
-        eko.output.manipulate.to_evol(operators)
-    elif not np.allclose(operators.rotations.inputpids, br.evol_basis_pids):
+    if np.allclose(operators.rotations.inputpids, br.flavor_basis_pids):
+        eko.io.manipulate.to_evol(operators)
+    # Here we are checking if the EKO contains the rotation matrix (flavor to evol)
+    elif not np.allclose(operators.rotations.inputpids, br.rotate_flavor_to_evolution):
         raise ValueError("The EKO is neither in flavor nor in evolution basis.")
     # do it
     order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
-    # TODO this is a hack to not break the CLI
-    # the problem is that the EKO output still does not contain the theory/operators card and
-    # so I can't compute alpha_s *here* if xir != 1
-    if np.isclose(xir, 1.0) and alphas_values is None:
-        mur2_grid = np.array(list(operators.Q2grid))
-        alphas_values = [astrong.a_s(q2) for q2 in operators.Q2grid]
+    # This is really the facto scale grid only for scheme A and C
+    muf2_grid = operators.mu2grid
+    # PineAPPL wants alpha_s = 4*pi*a_s
+    # remember that we already accounted for xif in the opcard generation
+    tcard = operators.theory_card
+    opcard = operators.operator_card
+    evmod = eko.couplings.couplings_mod_ev(opcard.configs.evolution_method)
+    # Couplings ask for the square of the masses
+    quark_masses = [(x.value) ** 2 for x in tcard.quark_masses]
+    sc = eko.couplings.Couplings(
+        tcard.couplings,
+        tcard.order,
+        evmod,
+        quark_masses,
+        hqm_scheme=tcard.quark_masses_scheme,
+        thresholds_ratios=tcard.matching,
+    )
+    # If we are computing scheme A we also need to pass fact_scale.
+    sv_method = None
+    if operators.operator_card.configs.scvar_method is not None:
+        sv_method = operators.operator_card.configs.scvar_method.name
+    # To compute the alphas values we are first reverting the factorization scale shift
+    # and then obtaining the renormalization scale using xir.
+    alphas_values = [
+        4.0
+        * np.pi
+        * sc.a_s(
+            xir * xir * muf2 / xif / xif,
+            fact_scale=muf2 if sv_method == "EXPONENTIATED" else None,
+        )
```

Comment:
Just to mention that here we should pass also `nf_to`. This most likely will make no differences, but you never know in the future... however it requires having and `nf_grid` as well

Reply:
@giacomomagni are you sure we need an explicit `nf_to`? I believe it might be useful for matching scales, but for as long as we do not solve https://github.com/NNPDF/eko/issues/149 I believe it should not be possible to retrieve this information.

Would you mind split your comment into a dedicated issue? (possibly tracking EKO ones)

Reply:
okay I can move the comment to issues